### PR TITLE
wallet: Initialise ThreadNotifyWallets with wallets best block

### PR DESCRIPTION
**Pull request by str4d**
The previous code assumed that the last chain tip notified to the wallet was equal to the node's chain tip at startup. However, this assumption fails if the node shuts down uncleanly, or if a wallet file is moved from one node to another.

We now try to start notifying from the wallet's best block, and if the node doesn't have that block we fall back to the node's chain tip like before.

Closes zcash/zcash#5805.

### CODE REVIEW

File: src/init.cpp [hunk 1]
```diff
@@ -1707,15 +1707,33 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
...
+        // an hour of chain sync older than the wallet's tip. We assume here
+        // that there is only a single wallet connected to the validation
+        // interface, which is currently true.
+#ifdef ENABLE_WALLET
+        if (pwalletMain)
+        {
+            LOCK2(cs_main, pwalletMain->cs_wallet);
+            const auto walletBestBlock = pwalletMain->GetBestBlock();
+            if (walletBestBlock != nullptr) {
+                pindexLastTip = walletBestBlock;
```
Comment (nuttycom) [inline src/init.cpp:1732]: In looking at uses of `pindexLastTip`, I think that we are missing some Orchard merkle tree handling here: https://github.com/zcash/zcash/pull/5809/files#diff-cd341be4887e632ceba211aee03c948d964396f7aed0146435163844fbcbad42R164-L154

Comment (ebfull) [reply src/init.cpp:1732]: @nuttycom I can't tell what you're linking to, that link is bad or something.

Comment (nuttycom) [reply src/init.cpp:1732]: Sorry, here's a better link: https://github.com/zcash/zcash/blob/nu5-consensus/src/validationinterface.cpp#L139-L154

**Review by nuttycom — CHANGES_REQUESTED** [review]
Looks like we need Orchard handling in https://github.com/zcash/zcash/blob/nu5-consensus/src/validationinterface.cpp#L139-L154 - does that belong in this PR?

File: src/init.cpp (hunk 1 above)
Comment (str4d) [reply src/init.cpp:1732]: Aha, I knew I'd seen it: we worked on it in a pairing for your PR #5789.

Comment (nuttycom) [reply src/init.cpp:1732]: Good grief, I clearly am losing it.

**Review by nuttycom — APPROVED** [review]
utACK https://github.com/zcash/zcash/pull/5809/commits/098a70ed896e1f5f8e5496de5828ddd5ea80c0df

**Review by therealyingtong — APPROVED** [review]
utACK

**Pull request merged**